From f4a2cda8fe3ca404690ddb509d46d66fe81b30da Mon Sep 17 00:00:00 2001
From: Matthew Rosato <mjrosato@linux.ibm.com>
Date: Tue, 26 Jan 2021 10:24:34 -0500
Subject: [PATCH 1/6] linux-headers: update against 5.10-rc1

commit 3650b228f83adda7e5ee532e2b90429c03f7b9ec

Signed-off-by: Matthew Rosato <mjrosato@linux.ibm.com>
[aw: drop pvrdma_ring.h changes to avoid revert of d73415a31547 ("qemu/atomic.h: rename atomic_ to qatomic_")]
Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
[mr: Backport: reduce header sync to only what is required for DMA limit fix]
Signed-off-by: Matthew Rosato <mjrosato@linux.ibm.com>

Origin: backport, https://git.qemu.org/?p=qemu.git;a=commit;h=53ba2eee52bff5a746e96835539a1079f6bcadd1
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1913395
Last-Update: 2021-01-27

---
 linux-headers/linux/vfio.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

--- a/linux-headers/linux/vfio.h
+++ b/linux-headers/linux/vfio.h
@@ -748,6 +748,21 @@ struct vfio_iommu_type1_info_cap_iova_ra
 	struct	vfio_iova_range iova_ranges[];
 };
 
+/*
+ * The DMA available capability allows to report the current number of
+ * simultaneously outstanding DMA mappings that are allowed.
+ *
+ * The structure below defines version 1 of this capability.
+ *
+ * avail: specifies the current number of outstanding DMA mappings allowed.
+ */
+#define VFIO_IOMMU_TYPE1_INFO_DMA_AVAIL 3
+
+struct vfio_iommu_type1_info_dma_avail {
+       struct  vfio_info_cap_header header;
+       __u32   avail;
+};
+
 #define VFIO_IOMMU_GET_INFO _IO(VFIO_TYPE, VFIO_BASE + 12)
 
 /**
